src/tests/testClpMap.h

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright (C) 1996-2023 The Squid Software Foundation and contributors
+ *
+ * Squid software is distributed under GPLv2+ license and includes
+ * contributions from numerous individuals and organizations.
+ * Please see the COPYING and CONTRIBUTORS files for details.
+ */
+
+#ifndef SQUID_BASE_TESTCLPMAP_H
+#define SQUID_BASE_TESTCLPMAP_H
+
+#include "compat/cppunit.h"
+#include "base/ClpMap.h"
+
+class testClpMap: public CPPUNIT_NS::TestFixture
+{
+    CPPUNIT_TEST_SUITE( testClpMap );
+    CPPUNIT_TEST( testMemoryCounter );
+    CPPUNIT_TEST( testConstructor );
+    CPPUNIT_TEST( testEntries );
+    CPPUNIT_TEST( testPutGetDelete );
+    CPPUNIT_TEST( testSetMemLimit );
+    CPPUNIT_TEST( testTtlExpiration );
+    CPPUNIT_TEST_SUITE_END();
+
+protected:
+    using TestMap = ClpMap<std::string, int>;
+
+    // add a standard set of elements to a map
```

Comment:
This is a convenience function, meant to add some elements to an array. It is not an API, it's local to the translation unit this file refers to.
Let's keep the complexity of the solution (and the number of iterations here) in line with the complexity of the problem; we are treating this as if it were a core API.
Same thing for Doxygen: do we even WANT to doxygen-document a two-line unit test helper function?

Reply:
FYI: I prefer to put these in the `private:` section at end of the class. It also helps to clarify that it is not a strangely named test method that got omitted from the `CPPUNIT_TEST()` list.

As for documenting; yes it is worthwhile - if only to assist future dev updating the unit internals.

Reply:
> As for documenting; yes it is worthwhile - if only to assist future dev updating the unit internals.

FWIW, I wasted at least 10, possibly 15 minutes because I did not understand the now-documented (thank you!) caveats of this complex (in its effects, not implementation) helper function.

Reply:
> Same thing for Doxygen: do we even WANT to doxygen-document a two-line unit test helper function?

IMO, yes, because if we do not use Doxygen, somebody will see a non-doxygen comment (e.g., when grepping sources for some specific name) and decide that this is how all (including non-test) functions are documented in Squid, and we will need to ask them to adjust their comments when they post a PR. Or somebody will start arguing whether a particular piece of code is a unit test helper function! Or that a Doxygen comment should be _removed_ from testing code because the latest testing code does not use Doxygen! Etc. Its better to be consistent when adding new code. Fortunately, it does not cost much extra time or effort to add a slash.